src/sources/file_descriptors/mod.rs

```diff
@@ -211,10 +211,12 @@ fn outputs(
     decoding: &DeserializerConfig,
     source_name: &'static str,
 ) -> Vec<Output> {
-    let legacy_host_key = host_key
-        .clone()
-        .and_then(|k| k.path)
-        .map(LegacyKey::InsertIfEmpty);
+    let legacy_host_key = Some(LegacyKey::InsertIfEmpty(
+        host_key
+            .clone()
+            .and_then(|k| k.path)
+            .unwrap_or_else(|| owned_value_path!(log_schema().host_key())),
```

Comment:
I this would break nested paths?

Reply:
yep. This assumes the host key is a single segment. You should use `parse_value_path(..)` instead. This is something I covered a bit in the new lookup docs I'm adding: https://github.com/vectordotdev/vector/pull/16717
and once more lookup migration is done this should be better typed (not a string) so it's harder to use it incorrectly